### PR TITLE
[Saas Connectors] Segment Masking [#354]

### LINKED ISSUE #354: [SaaS Connector] Segment Masking Test
### Blocked by

https://github.com/ethyca/fidesops/pull/350 and https://github.com/ethyca/fidesops/pull/349

### Is your feature request related to a specific problem?

- Add an end-to-end integration test for running a Segment masking request.  This uses a GDPR-delete endpoint.  I'd verify that the erasure request is submitted successfully, but on Segment's end, they have thirty days to delete the data, so  I don't think we can verify the data has been erased.  In testing, I submitted a request two days ago and it still hasn't processed.
- Segment saas config for deleting user data is included in #350

### PR DESCRIPTION
**Pull request by pattisdr**
# Purpose

Adds demonstration of a masking for third party integration: Segment.  Rather than issuing individual updates, a `data_protection_request` is defined in the yaml that issues something like a GDPR delete request that removes all data, regardless of what is defined on the Policy.

# Changes

- Fixes some bugs in the Segment `data_protection_request` that was introduced in Segment access request work in #350 
- Adds additional segment secrets for creating test user data
- Adds a segment erasure end-to-end integration test that creates user data, and events, runs an access request (which also makes sure that user data is there) and then runs an erasure.  The test confirms that the erasure was attempted, but we cannot verify the data is erased here due to the delays on the segment side (erasures I submitted 4-5 days ago have not completed yet)
- Updates Segment postman collection

# Checklist

- [x] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md))
- [x] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [x] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #354

### CODE REVIEW

File: tests/integration_tests/saas/test_segment_task.py [hunk 1]
```diff
@@ -131,3 +137,165 @@ def test_segment_saas_access_request_task(
...
+    segment_connection_config,
+    segment_dataset_config,
+    segment_identity_email,
+    segment_secrets,
+) -> None:
+    """Full erasure request based on the Segment SaaS config"""
+    config.execution.MASKING_STRICT = False  # Allow GDPR Delete
+
+    # Create user for GDPR delete
+    erasure_email = create_segment_test_data(segment_secrets, segment_identity_email)
```
Comment (pattisdr) [inline tests/integration_tests/saas/test_segment_task.py:227]: Slight concerns in running this too frequently. To test a GDPR delete I am creating a user and then issuing the masking request. None of the segment masking requests I've made over the last several days have completed, as Segment completes them in batches. Does our account have limits on this?

Historically it looks like we were running these tests once a week or so.

File: data/saas/config/segment_config.yml [hunk 2]
```diff
@@ -7,10 +7,13 @@ saas_config:
   connector_params:
     - name: domain
     - name: personas_domain
+    - name: api_domain
     - name: workspace
     - name: access_token
     - name: namespace_id
     - name: access_secret
+    - name: user_token
```
Comment (pattisdr) [inline data/saas/config/segment_config.yml:15]: I think I can actually remove these two pieces from the yaml, they are just for creating the user data for testing in the first place.

File: tests/integration_tests/saas/test_segment_task.py (hunk 1 above)
Comment (adriaaaa) [reply tests/integration_tests/saas/test_segment_task.py:227]: @eastandwestwind and @pattisdr according to segment's [API docs](https://segment.com/docs/privacy/user-deletion-and-suppression/), for the User Deletion and Suppression API: “[Contact Support](https://segment.com/help/contact/) if you need to process more than 100,000 users within a 30 day period.”

I think submitted a few every day would still be ok.

Comment (pattisdr) [reply tests/integration_tests/saas/test_segment_task.py:227]: thank you!

**Comment by pattisdr** [comment]
Getting this up to date ^

File: tests/integration_tests/saas/test_segment_task.py (hunk 1 above)
Comment (eastandwestwind) [reply tests/integration_tests/saas/test_segment_task.py:227]: yay, thanks!

**Review by eastandwestwind — APPROVED** [review]
looks great @pattisdr!

**Comment by pattisdr** [comment]
thanks for your review @eastandwestwind!

**Pull request merged**